### PR TITLE
autoid_service: make test TestGRPC stable when port 10080 conflicts

### LINKED ISSUE #43518: unstable test in the TestGRPC
## Bug Report

Please answer these questions before submitting your issue. Thanks!

### 3. What did you see instead (Required)

```
    logger.go:130: 2023-05-04T13:14:45.877+0800 INFO    grpc    [[core] [Channel #10] Channel switches to new LB policy "pick_first"]
    logger.go:130: 2023-05-04T13:14:45.877+0800 INFO    grpc    [[core] [Channel #10 SubChannel #12] Subchannel created]
    logger.go:130: 2023-05-04T13:14:45.877+0800 INFO    grpc    [[core] [Channel #10] Channel Connectivity change to CONNECTING]
    logger.go:130: 2023-05-04T13:14:45.877+0800 INFO    grpc    [[core] [Channel #10 SubChannel #12] Subchannel Connectivity change to CONNECTING]
    logger.go:130: 2023-05-04T13:14:45.877+0800 INFO    grpc    [[core] [Channel #10 SubChannel #12] Subchannel picks a new address "127.0.0.1:10080" to connect]
    logger.go:130: 2023-05-04T13:14:45.877+0800 INFO    grpc    [[core] pickfirstBalancer: UpdateSubConnState: 0xc018a3ddb8, {CONNECTING <nil>}]
    logger.go:130: 2023-05-04T13:14:45.878+0800 INFO    grpc    [[core] [Channel #10 SubChannel #12] Subchannel Connectivity change to READY]
    logger.go:130: 2023-05-04T13:14:45.878+0800 INFO    grpc    [[core] pickfirstBalancer: UpdateSubConnState: 0xc018a3ddb8, {READY <nil>}]
    logger.go:130: 2023-05-04T13:14:45.878+0800 INFO    grpc    [[core] [Channel #10] Channel Connectivity change to READY]
    autoid_test.go:201:
                Error Trace:    autoid_service/autoid_test.go:201
                Error:          Received unexpected error:
                                rpc error: code = Unimplemented desc = unknown service autoid.AutoIDAlloc
                Test:           TestGRPC
```

### PR DESCRIPTION
**Pull request by tiancaiamao**
### What problem does this PR solve?

Issue Number: close #43518

Problem Summary:

### Check List

Tests 

- [x] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

**Review by hawkingrei — APPROVED** [review]

File: autoid_service/autoid_test.go [hunk 1]
```diff
@@ -157,7 +157,8 @@ func TestGRPC(t *testing.T) {
 
 	var addr string
 	var listener net.Listener
-	for port := 10080; ; port++ {
```
Comment (xhebox) [inline autoid_service/autoid_test.go:160]: Can't we just listen `127.0.0.1:0` and use `listener.Addr()`?

**Review by xhebox — APPROVED** [review]

File: autoid_service/autoid_test.go [hunk 2]
```diff
@@ -156,15 +155,8 @@ func TestGRPC(t *testing.T) {
...
-		var err error
-		addr = fmt.Sprintf("127.0.0.1:%d", port)
-		listener, err = net.Listen("tcp", addr)
-		if err == nil {
-			break
-		}
-	}
+	listener, err := net.Listen("tcp", addr)
+	require.NoError(t, err)
 	defer listener.Close()
```
Comment (xhebox) [inline autoid_service/autoid_test.go:160]: Maybe we need to `addr = listener.Addr().String()`?

**Review by xhebox — CHANGES_REQUESTED** [review]

**Review by xhebox — APPROVED** [review]

File: autoid_service/autoid_test.go (hunk 2 above)
Comment (tiancaiamao) [reply autoid_service/autoid_test.go:159]: done

**Review by hawkingrei — APPROVED** [review]

**Pull request merged**